File: Move-Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int write = 0;
        for (int read = 0; read < nums.size(); read++) {
            if (nums[read] != 0) {
                nums[write] = nums[read];
                write++;
            }
        }
        for (int i = write; i < nums.size(); i++) {
            nums[i] = 0;
        }
    }
};

File: Move-Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        \\\
        :type nums: List[int]
        :rtype: None
        \\\
        write = 0
        for read in range(len(nums)):
            if nums[read] != 0:
                nums[write] = nums[read]
                write += 1
        for i in range(write, len(nums)):
            nums[i] = 0
